### PR TITLE
fix: Migrate to CircleCI

### PR DESCRIPTION
**Pull request by robcresswell**
The tests on Travis take over 40 minutes; migrate us to Circle for a more reasonable feedback loop

### CODE REVIEW

**Comment by dkontorovskyy** [comment]
Only question is: in Travis node 8 is default. You moved to be default 6. Is it for reason? In 3 weeks node 6 is reaching End-Of-Life.

**Comment by robcresswell** [comment]
@dkontorovskyy Node 6 is the lowest version we support, so it seems sensible to use the lowest version as default. Less chance of incompatible syntax being used somewhere.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,541 @@
...
+            pyenv install -s $PYTHON_VER_FULL
+            pyenv sh-shell $PYTHON_VER_FULL
+            python --version
+            export PATH=$HOME/.local/bin:$PATH
+            python -m pip install --user --quiet pip==$PIP_VER
+            # Call `pyenv rehash` to avoid issues after installing a new version of pip.
+            # see: https://github.com/pyenv/pyenv/issues/1141#issuecomment-383092896
+            pyenv sh-rehash
+            python -m pip --version
+            python -m pip install --user --quiet -r dev-requirements.txt --disable-pip-version-check
```
Comment (gitphill) [inline .circleci/config.yml]: is there a docker image we can use / make to avoid all this?

Comment (robcresswell) [reply .circleci/config.yml]: You'd need one per node / python combo, so 8 images? We absolutely could do this / should do this, but its an optimisation for later.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,541 @@
...
+          name: Use snyk-main npmjs user
+          command: echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" >> .npmrc
+      - run:
+          name: Install dependencies
+          command: npm install
+      - run:
+          name: Run tests
+          command: npm run build
+  release:
+    <<: *defaults
```
Comment (gitphill) [inline .circleci/config.yml]: travis has following note on release stage:

`node_js: "8" # This *has* to be the "build leader"`

we're using node:6 here - is that important?

Comment (darscan) [reply .circleci/config.yml]: > # This *has* to be the "build leader"

I originally wrote that.. it's fine, you can ignore it (was just a travis + semantic-release quirk)

File: .circleci/config.yml (hunk 1 above)
Comment (lili2311) [reply .circleci/config.yml]: This is what we do in the plugin https://github.com/snyk/snyk-python-plugin/blob/master/.travis.yml

File: .circleci/config.yml (hunk 2 above)
Comment (gitphill) [reply .circleci/config.yml]: oops sorry see you've already answered similar question from Daniel
yea think until we drop node 6 this make sense.

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,541 @@
+version: 2.1
+
+defaults: &defaults
+  docker:
+    - image: node:6
```
Comment (lili2311) [inline .circleci/config.yml]: pretty sure we move to 8 by default now? I think we can ditch 8 here

File: .circleci/config.yml (hunk 1 above)
Comment (robcresswell) [reply .circleci/config.yml]: @lili2311 I dont understand what you're saying :smile:

File: .circleci/config.yml (hunk 3 above)
Comment (robcresswell) [reply .circleci/config.yml]: @lili2311 CLI still supports Node 6. Your team hasn't dropped this yet, we discussed this recently :smile:

File: .circleci/config.yml (hunk 1 above)
Comment (darscan) [reply .circleci/config.yml]: lol, this is the plugin :D

File: .circleci/config.yml (hunk 3 above)
Comment (lili2311) [reply .circleci/config.yml]: ah derp this is the plugin ofc

Comment (lili2311) [reply .circleci/config.yml]: there is a plan in motion

Comment (robcresswell) [reply .circleci/config.yml]: Great!

File: .circleci/config.yml (hunk 1 above)
Comment (lili2311) [reply .circleci/config.yml]: oh dear I'm tired 😴

**Pull request merged**